llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -306,6 +306,23 @@ bool GCNTTIImpl::hasBranchDivergence(const Function *F) const {
   return !F || !ST->isSingleLaneExecution(*F);
 }
 
+unsigned GCNTTIImpl::getNumberOfParts(Type *Tp) {
+  // For certain 8 bit ops, we can pack a v4i8 into a single part
+  // (e.g. v4i8 shufflevectors -> v_perm v4i8, v4i8). Thus, we
```

Comment:
For certain ops, but this is much broader. It's only a few data flow ops where this might help, but this is making the costs of other operations too optimistic. We should avoid vectorizing non-dataflow operations.

Reply:
Ultimately, I still think it is the better approach to have SLP handle this type of vectorization (as opposed to writing a data-flow vectorizer). I think it should be left up to the cost model to accurately measure the quality and filter all but those few ops we are interested in. That was the intent of this PR, but it seems I'll need to tune the costs for calling convetion at least.

Reply:
I've added https://github.com/llvm/llvm-project/pull/113002 to vectorize i8s with a more refined cost model (instead of using shuffle vectorization to capture the usecase we are interested in)